packages/protocol/contracts/common/UsingPrecompiles.sol

```diff
@@ -132,12 +158,88 @@ contract UsingPrecompiles {
     public
+    view
     returns (bool)
   {
     bool success;
-    (success, ) = PROOF_OF_POSSESSION.call.value(0).gas(gasleft())(
-      abi.encodePacked(sender, blsKey, blsPop)
-    );
+    (success, ) = PROOF_OF_POSSESSION.staticcall(abi.encodePacked(sender, blsKey, blsPop));
     return success;
   }
+
+  /**
+   * @notice Parses block number out of header.
+   * @param header RLP encoded header
+   * @return Block number.
+   */
+  function getBlockNumberFromHeader(bytes memory header) public view returns (uint256) {
+    bytes memory out;
+    bool success;
+    (success, out) = BLOCK_NUMBER_FROM_HEADER.staticcall(abi.encodePacked(header));
+    require(success);
+    return getUint256FromBytes(out, 0);
+  }
+
+  /**
+   * @notice Computes hash of header.
+   * @param header RLP encoded header
+   * @return Header hash.
+   */
+  function hashHeader(bytes memory header) public view returns (bytes32) {
+    bytes memory out;
+    bool success;
+    (success, out) = HASH_HEADER.staticcall(abi.encodePacked(header));
+    require(success);
+    return getBytes32FromBytes(out, 0);
+  }
+
+  /**
+   * @notice Gets the parent seal bitmap from the header at the given block number.
+   * @param blockNumber Block number to retrieve. Must be within 4 epochs of the current number.
+   * @return Bitmap parent seal with set bits at indices correspoinding to signing validators.
+   */
+  function getParentSealBitmap(uint256 blockNumber) public view returns (bytes32) {
+    bytes memory out;
+    bool success;
+    (success, out) = GET_PARENT_SEAL_BITMAP.staticcall(abi.encodePacked(blockNumber));
+    require(success);
+    return getBytes32FromBytes(out, 0);
+  }
+
+  /**
+   * @notice Verifies the given header and returns the seal bitmap is successful.
+   * @param header RLP encoded header
+   * @return Bitmap parent seal with set bits at indices correspoinding to signing validators.
+   */
+  function getVerifiedSealBitmapFromHeader(bytes memory header) public view returns (bytes32) {
+    bytes memory out;
+    bool success;
+    (success, out) = GET_VERIFIED_SEAL_BITMAP.staticcall(abi.encodePacked(header));
+    require(success);
+    return getBytes32FromBytes(out, 0);
+  }
+
+  /**
+   * @notice Converts bytes to uint256.
+   * @param bs byte[] data
+   * @param start offset into byte data to convert
+   * @return uint256 data
+   */
+  function getUint256FromBytes(bytes memory bs, uint256 start) internal pure returns (uint256) {
+    return uint256(getBytes32FromBytes(bs, start));
+  }
+
+  /**
+   * @notice Converts bytes to bytes32.
+   * @param bs byte[] data
+   * @param start offset into byte data to convert
+   * @return bytes32 data
+   */
+  function getBytes32FromBytes(bytes memory bs, uint256 start) internal pure returns (bytes32) {
+    require(bs.length >= start + 32, "slicing out of range");
+    bytes32 x;
+    assembly {
+      x := mload(add(bs, add(0x20, start)))
```

Comment:
Can you do `add(bs, 32 + start)`? IMO the decimal representation is more readable